### ISSUE TITLE
[a11y] Switches should have on/off as value

### ISSUE THREAD

**Issue by goderbauer** [flutter/flutter#16395]
They should not be announced as selected/unselected.

**Comment by jonahwilliams** [comment]
While there are issues with the iOS implementation, in general we will need to solve this by identifying switches either as distinct from checkboxes or as a subtype of checkboxes.

For the former, we can do this by adding two new semantics flags - `HasToggleState` and `isToggled`.  For the latter we could only add a `HasToggledState` flag as a refinement of the `HasCheckedState`.

Thoughts @goderbauer ?

**Comment by goderbauer** [comment]
I would probably go with the pair of flags to stay within the current naming scheme. For the other option it wouldn't really be clear that HasToggledState belongs to HasCheckedState.

**Comment by emerssso** [comment]
Is there any timeline for this to be fixed for iOS?

**Comment by clocksmith** [comment]
Now that `toggled` is in Semantics, is this just a matter of setting this to true when the CupertinoSwitch is switched on?

**Comment by TahaTesser** [comment]
On testing on iOS with voice over enabled, it's announced as "Switch. button on and double tap to toggle settings" when  using `CupertinoSwitch` and `Switch`

<details> 
<summary>Code Sample</summary> 

```
import 'package:flutter/cupertino.dart';
import 'package:flutter/material.dart';

void main() => runApp(MyApp());

class MyApp extends StatelessWidget {
  @override
  Widget build(BuildContext context) {
    return MaterialApp(
      title: 'Material App',
      theme: ThemeData.dark(),
      home: Home(),
    );
  }
}

class Home extends StatefulWidget {
  @override
  _HomeState createState() => _HomeState();
}

class _HomeState extends State<Home> {
  bool _value = false;

  @override
  Widget build(BuildContext context) {
    return Scaffold(
      appBar: AppBar(
        title: Text('Material App Bar'),
      ),
      body: Center(
        child: CupertinoSwitch(
          value: _value,
          onChanged: (bool newValue) {
            setState(() {
              _value = newValue;
            });
          },
        ),
      ),
      floatingActionButton:
          FloatingActionButton(child: Icon(Icons.add), onPressed: () {}),
    );
  }
}

``` 

</details> 

<details> 
<summary>flutter doctor -v</summary> 

```
[✓] Flutter (Channel dev, 1.21.0-1.0.pre, on Mac OS X 10.15.6 19G73, locale
    en-GB)
    • Flutter version 1.21.0-1.0.pre at /Users/taha/Code/flutter_dev
    • Framework revision f25bd9c55c (8 days ago), 2020-07-14 20:26:01 -0400
    • Engine revision 99c2b3a245
    • Dart version 2.9.0 (build 2.9.0-21.0.dev 20bf2fcf56)

 
[✓] Android toolchain - develop for Android devices (Android SDK version 29.0.3)
    • Android SDK at /Users/taha/Code/sdk
    • Platform android-29, build-tools 29.0.3
    • ANDROID_HOME = /Users/taha/Code/sdk
    • Java binary at: /Applications/Android
      Studio.app/Contents/jre/jdk/Contents/Home/bin/java
    • Java version OpenJDK Runtime Environment (build
      1.8.0_242-release-1644-b3-6222593)
    • All Android licenses accepted.

[✓] Xcode - develop for iOS and macOS (Xcode 11.6)
    • Xcode at /Applications/Xcode.app/Contents/Developer
    • Xcode 11.6, Build version 11E708
    • CocoaPods version 1.9.3

[✓] Chrome - develop for the web
    • Chrome at /Applications/Google Chrome.app/Contents/MacOS/Google Chrome

[✓] Android Studio (version 4.0)
    • Android Studio at /Applications/Android Studio.app/Contents
    • Flutter plugin version 47.1.2
    • Dart plugin version 193.7361
    • Java version OpenJDK Runtime Environment (build
      1.8.0_242-release-1644-b3-6222593)

[✓] VS Code (version 1.47.2)
    • VS Code at /Applications/Visual Studio Code.app/Contents
    • Flutter extension version 3.12.2

[✓] Connected device (4 available)
    • SM M305F (mobile) • 32003c30dc19668f • android-arm64  • Android 10 (API
      29)
    • macOS (desktop)   • macos            • darwin-x64     • Mac OS X 10.15.6
      19G73
    • Web Server (web)  • web-server       • web-javascript • Flutter Tools
    • Chrome (web)      • chrome           • web-javascript • Google Chrome
      84.0.4147.89

• No issues found!
```

</details> 

Closing, as fixed. If you disagree, please write in the comments and I will reopen it.
Thank you
